.editorconfig

```diff
@@ -0,0 +1,42 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.{json,toml,yml,gyp}]
+indent_style = space
+indent_size = 2
+
+[*.js]
+indent_style = space
+indent_size = 2
+
+[*.rs]
+indent_style = space
+indent_size = 4
+
+[*.{c,cc,h}]
+indent_style = space
+indent_size = 4
+
+[src/scanner.{c,cc}]
+indent_size = 2
```

Comment:
Ah thanks, I'll have a look. Most of this was autogenerated by tree-sitter from the upgrade, so I'm not sure where that discrepancy is coming from.

Reply:
Yes. Not sure why 4-space-indent is the official config since the generated parser uses 2 spaces (as do most scanners I've seen in the wild), but we'll just follow their convention.